site/docs/v1/tech/guides/advanced-registration-forms.adoc

```diff
@@ -706,6 +706,34 @@ Assume you've enabled the Facebook identity provider and allowed for registratio
 
 After a user signs up with Facebook, they'll be dropped back into the registration flow on the second step. They'll be required to complete the registration from from the second step onward before they are fully registered.
 
+=== Hidden Fields
+
+You may create form fields in an advanced registration form that capture information about a user, such as a referrer, using hidden fields.
+
+To do so:
+
+* create the form field. You can set the cont
+* add it to the form
+* modify your theme to not display the form field
+
+image::guides/advanced-registration-forms/hidden-field.png[Example of hidden form field.,width=1200,role=bottom-cropped]
+
+You'll have to modify the freemarker code which generates the registration form. The `customField` macro is what generates the HTML, so you could add something similar to
+
+[source,freemarker,title=Generating a hidden field]
+----
+[#if field.key == "user.data.referrer"]
+  <input id="${fieldId}" type="hidden" name="${key}">
+[/#if]
+----
+
+Set this form field value using JavaScript.
+
+[source,javascript,title=Setting a hidden field]
+----
+document.getElementById("user_data_referrer").value = "value";
```

Comment:
It isn't clear to me in this example that `${fieldId}` would be equal to `user_data_referrer`. For example's sake - would it be better to hard code an id for use in the JavaScript?

Reply:
Sure. Will hardcode and then merge.